src/main/java/uk/gov/homeoffice/digital/sas/balancecalculator/client/RestClient.java

```diff
@@ -0,0 +1,108 @@
+import uk.gov.homeoffice.digital.sas.balancecalculator.models.accrual.Accrual;
+import uk.gov.homeoffice.digital.sas.balancecalculator.models.accrual.Agreement;
+
+@Component
+public class RestClient {
+
+  public static final String TENANT_ID_STRING_IDENTIFIER = "tenantId";
+  public static final String FILTER_STRING_IDENTIFIER = "filter";
+  public static final String GET_ACCRUAL_BY_TYPE_AND_DATE_EXCEPTION = """
+  Non-unique Accrual result for tenantId: {0},
+   personId: {1}, accrualTypeId: {2} and accrualDate: {3}
+      """;
+  public static final String GET_AGREEMENT_BY_ID_EXCEPTION =
+      "Non-unique Accrual result for agreementId: '{0}'";
+
+  private final RestTemplate restTemplate;
+  private final String accrualsFilterUrl;
+  private final String agreementsByIdUrl;
+
+
+  @Autowired
+  public RestClient(RestTemplateBuilder builder,
+      @Value("${balance.calculator.accruals.url}") String accrualsUrl) {
+    this.restTemplate = builder.build();
+    this.accrualsFilterUrl =
+        accrualsUrl + "/resources/accruals?tenantId={tenantId}&filter={filter}";
+    this.agreementsByIdUrl =
+        accrualsUrl + "/resources/agreements/{agreementId}?tenantId={tenantId}";
+  }
+
+  public Accrual getAccrualByTypeAndDate(String tenantId, String personId, String accrualTypeId,
+      LocalDate accrualDate) {
+    Map<String, String> parameters = Map.of(
+        TENANT_ID_STRING_IDENTIFIER, tenantId,
+        FILTER_STRING_IDENTIFIER,
+        "personId=='" + personId + "'"
+            + "&&accrualTypeId=='" + accrualTypeId + "'"
+            + "&&accrualDate=='" + accrualDate + "'"
+    );
+
+    ResponseEntity<ApiResponse<Accrual>> entity
+        = restTemplate.exchange(accrualsFilterUrl, HttpMethod.GET, null,
+          new ParameterizedTypeReference<>() {}, parameters);
+
+    if (Objects.requireNonNull(entity.getBody()).getItems().size() == 1) {
+      return Objects.requireNonNull(entity.getBody()).getItems().get(0);
+    } else {
+      throw new NonUniqueResultException(
+          MessageFormat.format(GET_ACCRUAL_BY_TYPE_AND_DATE_EXCEPTION,
+              tenantId, personId, accrualTypeId, accrualDate));
+    }
+  }
+
+  public Agreement getAgreementById(String tenantId, String agreementId) {
+    Map<String, String> parameters = Map.of(TENANT_ID_STRING_IDENTIFIER, tenantId,
+        "agreementId", agreementId);
+
+    ResponseEntity<ApiResponse<Agreement>> entity
+        = restTemplate.exchange(agreementsByIdUrl, HttpMethod.GET, null,
+          new ParameterizedTypeReference<>() {}, parameters);
+
+    if (Objects.requireNonNull(entity.getBody()).getItems().size() == 1) {
+      return Objects.requireNonNull(entity.getBody()).getItems().get(0);
+    } else {
+      throw new NonUniqueResultException(
+          MessageFormat.format(GET_AGREEMENT_BY_ID_EXCEPTION, agreementId));
+    }
+  }
+
+  public List<Accrual> getAccrualsBetweenDates(String tenantId, String personId,
```

Comment:
Unless we choose to get all of the accruals upfront, should this not take the accrual type as well.

Reply:
Kept the accrual type out of the arguments to minimises trips to REST API